### PR TITLE
export `IdSet` and document it

### PR DESCRIPTION
**Pull request by KristofferC**
It feels kind of odd to have `IdDict` exported and documented and not `IdSet`.

### CODE REVIEW

File: base/idset.jl [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
 
+"""
+    IdSet{T}([itr])
+    IdSet()
+
+IdSet{T}() constructs a set (see [Set](@ref)) using
+`===` as equality with values of type `V`.
+
+In the example below, the values are all `isequal` so they get overwritten.
+The `IdDict` compares by `===` so preserves the 3 different keys.
```
[the remark this answers is not in the document]
Comment (KristofferC) [reply base/idset.jl:11]: 🤦‍♂️

File: base/idset.jl [hunk 2]
```diff
@@ -1,5 +1,28 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+    IdSet{T}([itr])
+    IdSet()
+
+IdSet{T}() constructs a set (see [Set](@ref)) using
```
Comment (inkydragon) [inline base/idset.jl:7]: fix CI error
```suggestion
IdSet{T}() constructs a set (see [`Set`](@ref)) using
```

Comment (KristofferC) [reply base/idset.jl]: <3

File: base/idset.jl [hunk 3]
```diff
@@ -1,5 +1,28 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+    IdSet{T}([itr])
+    IdSet()
+
+IdSet{T}() constructs a set (see [`Set`](@ref)) using
+`===` as equality with values of type `V`.
```
Comment (halleysfifthinc) [inline base/idset.jl:8]: Typo?
```suggestion
`===` as equality with values of type `T`.
```

Comment (KristofferC) [reply base/idset.jl:8]: Yes :(

**Pull request merged**